server/src/main/kotlin/suwayomi/tachidesk/global/controller/GlobalMetaController.kt

```diff
@@ -0,0 +1,53 @@
+package suwayomi.tachidesk.global.controller
+
+/*
+ * Copyright (C) Contributors to the Suwayomi project
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import io.javalin.http.HttpCode
+import suwayomi.tachidesk.global.impl.GlobalMeta
+import suwayomi.tachidesk.server.util.formParam
+import suwayomi.tachidesk.server.util.handler
+import suwayomi.tachidesk.server.util.withOperation
+
+object GlobalMetaController {
+    /** used to modify a category's meta parameters */
+    val getMeta = handler(
+        documentWith = {
+            withOperation {
+                summary("Server level meta mapping")
+                description("Get a list of globally stored key-value mapping, you can set values for whatever you want inside it.")
+            }
+        },
+        behaviorOf = { ctx ->
+            ctx.json(GlobalMeta.getMetaMap())
+            ctx.status(200)
+        },
+        withResults = {
+            httpCode(HttpCode.OK)
+        }
+    )
+
+    /** used to modify global meta parameters */
+    val modifyMeta = handler(
+        formParam<String>("key"),
+        formParam<String>("value"),
+        documentWith = {
+            withOperation {
+                summary("Add meta data to the global meta mapping")
+                description("A simple Key-Value stored at server global level, you can set values for whatever you want inside it.")
+            }
+        },
+        behaviorOf = { ctx, key, value ->
+            GlobalMeta.modifyMeta(key, value)
```

Comment:
Would it be useful to be able to edit multiple keys in one request?

Reply:
Maybe in a future PR

Reply:
> If it may be added in future, it might be good idea to make it future-proof now. As it is proposed would make it breaking change when changed later.

It's fine as long as we haven't made a stable release.

> Also I have noticed that some places in API use JSON body and some places use fromParams. Is there any system in which uses which? I find JSON more practical, since we don't use html forms anywhere.

I prefer pathParam and formParam while the request expects a few keys, when it takes a lot of or long parameters I use JSON.